pkg/media/media.go

```diff
@@ -198,6 +198,7 @@ var ErrInvalidMetadata = errors.New("invalid segment metadata")
 
 func ParseSegmentAssertions(ctx context.Context, mani *manifeststore.Manifest) (*SegmentMetadata, error) {
 	ctx, span := otel.Tracer("signer").Start(ctx, "ParseSegmentAssertions")
+	defer ctx.Done()
```

Comment:
This one confuses me, why do we need to wait on context.Done() after this function ends?